DCS_pcall.lua

```diff
@@ -17,127 +34,1227 @@ DCS = {
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        getDispatcherTowerPos = function(_airbase)
+            local _status, _response = pcall(function()
+                return _airbase:getDispatcherTowerPos();
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        getRadioSilentMode = function(_airbase)
+            local _status, _response = pcall(function()
+                return _airbase:getRadioSilentMode();
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        setRadioSilentMode = function(_airbase,_silent)
+            local _status, _response = pcall(function()
+                return _airbase:setRadioSilentMode(_silent or false);
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        autoCapture = function(_airbase,_setting)
+            local _status, _response = pcall(function()
+                return _airbase:autoCapture(_setting or true);
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        autoCaptureIsOn = function(_airbase)
+            local _status, _response = pcall(function()
+                return _airbase:autoCaptureIsOn();
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        setCoalition = function(_airbase, _coalition)
+            local _status, _response = pcall(function()
+                return _airbase:setCoalition(_coalition or coalition.side.NEUTRAL);
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end,
+
+        getWarehouse = function(_airbase)
+            local _status, _response = pcall(function()
+                return _airbase:getWarehouse();
+            end)
+            if _status then
+                return _response;
+            else
+                -- TODO Need logging added here
+                return nil;
+            end
+        end
+    },
+
+    atmosphere = {
+        getWind = function(_vec3)
+            local _status, _response = pcall(function()
+                return atmosphere.getWind(_vec3);
```

Comment:
this should probably get from `DCS.proxy["atmosphere"]` instead of going direct to `atmosphere`

Reply:
yeah, i did that before finding out about the proxies you did. If you haven't fixed it i will sort it out next time i am adding code.

Reply:
I did it already 😄 - I "resolved" the comment, just didn't reply back about it. My bad.